### ISSUE TITLE
tests of fluent-plugin-secure-forward fails with fluentd v0.14.0.rc.1 (undefined method `level=' for class `Fluent::Test::DummyLogDevice')

### ISSUE THREAD

**Issue by cosmo0920** [fluent/fluentd#976]
### How to reproduce

```bash
$ git@github.com:tagomoris/fluent-plugin-secure-forward.git
$ cd fluent-plugin-secure-forward
```

Append the following line into Gemfile:

```bash
gem "fluentd", "= 0.14.0.rc.1"
```

Run tests:

```
$ bundle exec rake test
```

### Result

Tests couldn't start.

```log
$ bundle exec rake test
/Users/hhatake/.rbenv/versions/2.3.0/bin/ruby -w -I"lib:lib:test" -I"/Users/hhatake/Github/fluent-plugin-secure-forward/vendor/bundle/ruby/2.3.0/gems/rake-11.1.2/lib" "/Users/hhatake/Github/fluent-plugin-secure-forward/vendor/bundle/ruby/2.3.0/gems/rake-11.1.2/lib/rake/rake_test_loader.rb" "test/**/test_*.rb" 
/Users/hhatake/Github/fluent-plugin-secure-forward/vendor/bundle/ruby/2.3.0/gems/cool.io-1.4.3/lib/cool.io/loop.rb:41: warning: method redefined; discarding old initialize
(eval):1: warning: method redefined; discarding old on_read
/Users/hhatake/Github/fluent-plugin-secure-forward/vendor/bundle/ruby/2.3.0/gems/cool.io-1.4.3/lib/cool.io/io.rb:79: warning: previous definition of on_read was here
(eval):1: warning: method redefined; discarding old on_write_complete
/Users/hhatake/Github/fluent-plugin-secure-forward/vendor/bundle/ruby/2.3.0/gems/cool.io-1.4.3/lib/cool.io/io.rb:83: warning: previous definition of on_write_complete was here
(eval):1: warning: method redefined; discarding old on_close
/Users/hhatake/Github/fluent-plugin-secure-forward/vendor/bundle/ruby/2.3.0/gems/cool.io-1.4.3/lib/cool.io/io.rb:87: warning: previous definition of on_close was here
(eval):1: warning: method redefined; discarding old on_readable
(eval):1: warning: method redefined; discarding old on_writable
(eval):1: warning: method redefined; discarding old on_timer
(eval):1: warning: method redefined; discarding old on_signal
/Users/hhatake/Github/fluent-plugin-secure-forward/vendor/bundle/ruby/2.3.0/gems/cool.io-1.4.3/lib/cool.io/async_watcher.rb:24: warning: previous definition of on_signal was here
(eval):1: warning: method redefined; discarding old on_connection
/Users/hhatake/Github/fluent-plugin-secure-forward/vendor/bundle/ruby/2.3.0/gems/cool.io-1.4.3/lib/cool.io/listener.rb:35: warning: previous definition of on_connection was here
/Users/hhatake/Github/fluent-plugin-secure-forward/vendor/bundle/ruby/2.3.0/gems/cool.io-1.4.3/lib/cool.io/dns_resolver.rb:171: warning: assigned but unused variable - ancount
(eval):1: warning: method redefined; discarding old on_success
/Users/hhatake/Github/fluent-plugin-secure-forward/vendor/bundle/ruby/2.3.0/gems/cool.io-1.4.3/lib/cool.io/dns_resolver.rb:89: warning: previous definition of on_success was here
(eval):1: warning: method redefined; discarding old on_failure
/Users/hhatake/Github/fluent-plugin-secure-forward/vendor/bundle/ruby/2.3.0/gems/cool.io-1.4.3/lib/cool.io/dns_resolver.rb:93: warning: previous definition of on_failure was here
/Users/hhatake/Github/fluent-plugin-secure-forward/vendor/bundle/ruby/2.3.0/gems/cool.io-1.4.3/lib/cool.io/socket.rb:30: warning: method redefined; discarding old attach
(eval):1: warning: previous definition of attach was here
(eval):1: warning: method redefined; discarding old on_connect
/Users/hhatake/Github/fluent-plugin-secure-forward/vendor/bundle/ruby/2.3.0/gems/cool.io-1.4.3/lib/cool.io/socket.rb:42: warning: previous definition of on_connect was here
(eval):1: warning: method redefined; discarding old on_connect_failed
/Users/hhatake/Github/fluent-plugin-secure-forward/vendor/bundle/ruby/2.3.0/gems/cool.io-1.4.3/lib/cool.io/socket.rb:46: warning: previous definition of on_connect_failed was here
/Users/hhatake/Github/fluent-plugin-secure-forward/vendor/bundle/ruby/2.3.0/gems/fluentd-0.14.0.rc.1/lib/fluent/log.rb:58:in `public_instance_method': undefined method `level=' for class `Fluent::Test::DummyLogDevice' (NameError)
	from /Users/hhatake/Github/fluent-plugin-secure-forward/vendor/bundle/ruby/2.3.0/gems/fluentd-0.14.0.rc.1/lib/fluent/log.rb:58:in `initialize'
	from /Users/hhatake/Github/fluent-plugin-secure-forward/test/helper.rb:16:in `new'
	from /Users/hhatake/Github/fluent-plugin-secure-forward/test/helper.rb:16:in `<top (required)>'
	from /Users/hhatake/Github/fluent-plugin-secure-forward/test/plugin/test_in_secure_forward.rb:1:in `require'
	from /Users/hhatake/Github/fluent-plugin-secure-forward/test/plugin/test_in_secure_forward.rb:1:in `<top (required)>'
	from /Users/hhatake/Github/fluent-plugin-secure-forward/vendor/bundle/ruby/2.3.0/gems/rake-11.1.2/lib/rake/rake_test_loader.rb:10:in `require'
	from /Users/hhatake/Github/fluent-plugin-secure-forward/vendor/bundle/ruby/2.3.0/gems/rake-11.1.2/lib/rake/rake_test_loader.rb:10:in `block (2 levels) in <main>'
	from /Users/hhatake/Github/fluent-plugin-secure-forward/vendor/bundle/ruby/2.3.0/gems/rake-11.1.2/lib/rake/rake_test_loader.rb:9:in `each'
	from /Users/hhatake/Github/fluent-plugin-secure-forward/vendor/bundle/ruby/2.3.0/gems/rake-11.1.2/lib/rake/rake_test_loader.rb:9:in `block in <main>'
	from /Users/hhatake/Github/fluent-plugin-secure-forward/vendor/bundle/ruby/2.3.0/gems/rake-11.1.2/lib/rake/rake_test_loader.rb:4:in `select'
	from /Users/hhatake/Github/fluent-plugin-secure-forward/vendor/bundle/ruby/2.3.0/gems/rake-11.1.2/lib/rake/rake_test_loader.rb:4:in `<main>'
rake aborted!
Command failed with status (1): [ruby -w -I"lib:lib:test" -I"/Users/hhatake/Github/fluent-plugin-secure-forward/vendor/bundle/ruby/2.3.0/gems/rake-11.1.2/lib" "/Users/hhatake/Github/fluent-plugin-secure-forward/vendor/bundle/ruby/2.3.0/gems/rake-11.1.2/lib/rake/rake_test_loader.rb" "test/**/test_*.rb" ]

Tasks: TOP => test
(See full trace by running task with --trace)
```

### Expected result

Tests succeeded without any changes.

**Comment by cosmo0920** [comment]
If I apply locally this patch:

```diff
diff --git a/test/output/out_forest_test.rb b/test/output/out_forest_test.rb
index db957b0..7b75067 100644
--- a/test/output/out_forest_test.rb
+++ b/test/output/out_forest_test.rb
@@ -1,4 +1,4 @@
-class Fluent::ForestTestOutput < Fluent::Output
+class Fluent::ForestTestOutput < Fluent::Plugin::BareOutput
   Fluent::Plugin.register_output('forest_test', self)
 
   config_param :key_name, :string, :default => 'msg'
```

I've got another test errors:

```log
 $ bundle exec rake test
/Users/hhatake/.rbenv/versions/2.3.0/bin/ruby -w -I"lib:lib:test" -I"/Users/hhatake/Github/fluent-plugin-forest/vendor/bundle/ruby/2.3.0/gems/rake-11.1.2/lib" "/Users/hhatake/Github/fluent-plugin-forest/vendor/bundle/ruby/2.3.0/gems/rake-11.1.2/lib/rake/rake_test_loader.rb" "test/**/test_*.rb" 
/Users/hhatake/Github/fluent-plugin-forest/test/plugin/test_out_forest.rb:36: warning: assigned but unused variable - d
/Users/hhatake/Github/fluent-plugin-forest/vendor/bundle/ruby/2.3.0/gems/cool.io-1.4.3/lib/cool.io/loop.rb:41: warning: method redefined; discarding old initialize
(eval):1: warning: method redefined; discarding old on_read
/Users/hhatake/Github/fluent-plugin-forest/vendor/bundle/ruby/2.3.0/gems/cool.io-1.4.3/lib/cool.io/io.rb:79: warning: previous definition of on_read was here
(eval):1: warning: method redefined; discarding old on_write_complete
/Users/hhatake/Github/fluent-plugin-forest/vendor/bundle/ruby/2.3.0/gems/cool.io-1.4.3/lib/cool.io/io.rb:83: warning: previous definition of on_write_complete was here
(eval):1: warning: method redefined; discarding old on_close
/Users/hhatake/Github/fluent-plugin-forest/vendor/bundle/ruby/2.3.0/gems/cool.io-1.4.3/lib/cool.io/io.rb:87: warning: previous definition of on_close was here
(eval):1: warning: method redefined; discarding old on_readable
(eval):1: warning: method redefined; discarding old on_writable
(eval):1: warning: method redefined; discarding old on_timer
(eval):1: warning: method redefined; discarding old on_signal
/Users/hhatake/Github/fluent-plugin-forest/vendor/bundle/ruby/2.3.0/gems/cool.io-1.4.3/lib/cool.io/async_watcher.rb:24: warning: previous definition of on_signal was here
(eval):1: warning: method redefined; discarding old on_connection
/Users/hhatake/Github/fluent-plugin-forest/vendor/bundle/ruby/2.3.0/gems/cool.io-1.4.3/lib/cool.io/listener.rb:35: warning: previous definition of on_connection was here
/Users/hhatake/Github/fluent-plugin-forest/vendor/bundle/ruby/2.3.0/gems/cool.io-1.4.3/lib/cool.io/dns_resolver.rb:171: warning: assigned but unused variable - ancount
(eval):1: warning: method redefined; discarding old on_success
/Users/hhatake/Github/fluent-plugin-forest/vendor/bundle/ruby/2.3.0/gems/cool.io-1.4.3/lib/cool.io/dns_resolver.rb:89: warning: previous definition of on_success was here
(eval):1: warning: method redefined; discarding old on_failure
/Users/hhatake/Github/fluent-plugin-forest/vendor/bundle/ruby/2.3.0/gems/cool.io-1.4.3/lib/cool.io/dns_resolver.rb:93: warning: previous definition of on_failure was here
/Users/hhatake/Github/fluent-plugin-forest/vendor/bundle/ruby/2.3.0/gems/cool.io-1.4.3/lib/cool.io/socket.rb:30: warning: method redefined; discarding old attach
(eval):1: warning: previous definition of attach was here
(eval):1: warning: method redefined; discarding old on_connect
/Users/hhatake/Github/fluent-plugin-forest/vendor/bundle/ruby/2.3.0/gems/cool.io-1.4.3/lib/cool.io/socket.rb:42: warning: previous definition of on_connect was here
(eval):1: warning: method redefined; discarding old on_connect_failed
/Users/hhatake/Github/fluent-plugin-forest/vendor/bundle/ruby/2.3.0/gems/cool.io-1.4.3/lib/cool.io/socket.rb:46: warning: previous definition of on_connect_failed was here
Loaded suite /Users/hhatake/Github/fluent-plugin-forest/vendor/bundle/ruby/2.3.0/gems/rake-11.1.2/lib/rake/rake_test_loader
Started
.E
=================================================================================================================================================================================================
Error: test_emit(ForestOutputTest): RuntimeError: BUG: use router.emit instead of Engine.emit
/Users/hhatake/Github/fluent-plugin-forest/vendor/bundle/ruby/2.3.0/bundler/gems/fluentd-d5476d747846/lib/fluent/engine.rb:132:in `emit'
/Users/hhatake/Github/fluent-plugin-forest/test/output/out_forest_test.rb:39:in `block in emit'
/Users/hhatake/Github/fluent-plugin-forest/vendor/bundle/ruby/2.3.0/bundler/gems/fluentd-d5476d747846/lib/fluent/event.rb:74:in `each'
/Users/hhatake/Github/fluent-plugin-forest/test/output/out_forest_test.rb:31:in `emit'
/Users/hhatake/Github/fluent-plugin-forest/lib/fluent/plugin/out_forest.rb:171:in `emit'
/Users/hhatake/Github/fluent-plugin-forest/vendor/bundle/ruby/2.3.0/bundler/gems/fluentd-d5476d747846/lib/fluent/compat/output.rb:158:in `process'
/Users/hhatake/Github/fluent-plugin-forest/vendor/bundle/ruby/2.3.0/bundler/gems/fluentd-d5476d747846/lib/fluent/plugin/bare_output.rb:53:in `emit_sync'
/Users/hhatake/Github/fluent-plugin-forest/vendor/bundle/ruby/2.3.0/bundler/gems/fluentd-d5476d747846/lib/fluent/test/output_test.rb:46:in `emit'
/Users/hhatake/Github/fluent-plugin-forest/test/plugin/test_out_forest.rb:541:in `block in test_emit'
     538:   def test_emit
     539:     d = create_driver
     540:     time = Time.parse("2012-01-02 13:14:15").to_i
  => 541:     d.tag = 'test.first';  d.run { d.emit({'f' => "message 1"}, time) }
     542:     d.tag = 'test.second'; d.run { d.emit({'f' => "message 2"}, time) }
     543:     d.tag = 'test.foo.bar'; d.run { d.emit({'f' => "message 3"}, time) }
     544:     d.tag = 'test.foo.baz'; d.run { d.emit({'f' => "message 4"}, time) }
/Users/hhatake/Github/fluent-plugin-forest/vendor/bundle/ruby/2.3.0/bundler/gems/fluentd-d5476d747846/lib/fluent/test/input_test.rb:123:in `block in run'
/Users/hhatake/Github/fluent-plugin-forest/vendor/bundle/ruby/2.3.0/bundler/gems/fluentd-d5476d747846/lib/fluent/test/base.rb:81:in `run'
/Users/hhatake/Github/fluent-plugin-forest/vendor/bundle/ruby/2.3.0/bundler/gems/fluentd-d5476d747846/lib/fluent/test/input_test.rb:122:in `run'
/Users/hhatake/Github/fluent-plugin-forest/test/plugin/test_out_forest.rb:541:in `test_emit'
=================================================================================================================================================================================================
E
=================================================================================================================================================================================================
Error: test_faild_plant(ForestOutputTest): RuntimeError: BUG: use router.emit instead of Engine.emit
/Users/hhatake/Github/fluent-plugin-forest/vendor/bundle/ruby/2.3.0/bundler/gems/fluentd-d5476d747846/lib/fluent/engine.rb:132:in `emit'
/Users/hhatake/Github/fluent-plugin-forest/test/output/out_forest_test.rb:39:in `block in emit'
/Users/hhatake/Github/fluent-plugin-forest/vendor/bundle/ruby/2.3.0/bundler/gems/fluentd-d5476d747846/lib/fluent/event.rb:74:in `each'
/Users/hhatake/Github/fluent-plugin-forest/test/output/out_forest_test.rb:31:in `emit'
/Users/hhatake/Github/fluent-plugin-forest/lib/fluent/plugin/out_forest.rb:171:in `emit'
/Users/hhatake/Github/fluent-plugin-forest/vendor/bundle/ruby/2.3.0/bundler/gems/fluentd-d5476d747846/lib/fluent/compat/output.rb:158:in `process'
/Users/hhatake/Github/fluent-plugin-forest/vendor/bundle/ruby/2.3.0/bundler/gems/fluentd-d5476d747846/lib/fluent/plugin/bare_output.rb:53:in `emit_sync'
/Users/hhatake/Github/fluent-plugin-forest/vendor/bundle/ruby/2.3.0/bundler/gems/fluentd-d5476d747846/lib/fluent/test/output_test.rb:46:in `emit'
/Users/hhatake/Github/fluent-plugin-forest/test/plugin/test_out_forest.rb:507:in `block in test_faild_plant'
     504:   def test_faild_plant
     505:     d = create_driver
     506:     time = Time.parse("2012-01-02 13:14:15").to_i
  => 507:     d.tag = 'test.xxxxxx';  d.run { d.emit({'f' => "message 1"}, time) }
     508:     emits = d.emits
     509:     assert_equal 1, emits.length
     510: 
/Users/hhatake/Github/fluent-plugin-forest/vendor/bundle/ruby/2.3.0/bundler/gems/fluentd-d5476d747846/lib/fluent/test/input_test.rb:123:in `block in run'
/Users/hhatake/Github/fluent-plugin-forest/vendor/bundle/ruby/2.3.0/bundler/gems/fluentd-d5476d747846/lib/fluent/test/base.rb:81:in `run'
/Users/hhatake/Github/fluent-plugin-forest/vendor/bundle/ruby/2.3.0/bundler/gems/fluentd-d5476d747846/lib/fluent/test/input_test.rb:122:in `run'
/Users/hhatake/Github/fluent-plugin-forest/test/plugin/test_out_forest.rb:507:in `test_faild_plant'
=================================================================================================================================================================================================
...........E
=================================================================================================================================================================================================
Error: test_configure(ForestTestOutputTest): NoMethodError: undefined method `router=' for #<Fluent::ForestTestOutput:007f9f12f78bc8>
/Users/hhatake/Github/fluent-plugin-forest/vendor/bundle/ruby/2.3.0/bundler/gems/fluentd-d5476d747846/lib/fluent/test/base.rb:53:in `initialize'
/Users/hhatake/Github/fluent-plugin-forest/vendor/bundle/ruby/2.3.0/bundler/gems/fluentd-d5476d747846/lib/fluent/test/input_test.rb:25:in `initialize'
/Users/hhatake/Github/fluent-plugin-forest/vendor/bundle/ruby/2.3.0/bundler/gems/fluentd-d5476d747846/lib/fluent/test/output_test.rb:38:in `initialize'
/Users/hhatake/Github/fluent-plugin-forest/test/plugin/test_out_forest_test.rb:17:in `new'
/Users/hhatake/Github/fluent-plugin-forest/test/plugin/test_out_forest_test.rb:17:in `create_driver'
/Users/hhatake/Github/fluent-plugin-forest/test/plugin/test_out_forest_test.rb:21:in `test_configure'
     18:   end
     19: 
     20:   def test_configure
  => 21:     d = create_driver
     22:     assert_equal 'foo.bar', d.instance.tag
     23:     assert_equal 'fixed:', d.instance.prefix
     24:     assert_equal ':end', d.instance.suffix
=================================================================================================================================================================================================
E
=================================================================================================================================================================================================
Error: test_emit(ForestTestOutputTest): NoMethodError: undefined method `router=' for #<Fluent::ForestTestOutput:007f9f12f9a2c8>
/Users/hhatake/Github/fluent-plugin-forest/vendor/bundle/ruby/2.3.0/bundler/gems/fluentd-d5476d747846/lib/fluent/test/base.rb:53:in `initialize'
/Users/hhatake/Github/fluent-plugin-forest/vendor/bundle/ruby/2.3.0/bundler/gems/fluentd-d5476d747846/lib/fluent/test/input_test.rb:25:in `initialize'
/Users/hhatake/Github/fluent-plugin-forest/vendor/bundle/ruby/2.3.0/bundler/gems/fluentd-d5476d747846/lib/fluent/test/output_test.rb:38:in `initialize'
/Users/hhatake/Github/fluent-plugin-forest/test/plugin/test_out_forest_test.rb:17:in `new'
/Users/hhatake/Github/fluent-plugin-forest/test/plugin/test_out_forest_test.rb:17:in `create_driver'
/Users/hhatake/Github/fluent-plugin-forest/test/plugin/test_out_forest_test.rb:28:in `test_emit'
     25:   end
     26: 
     27:   def test_emit
  => 28:     d = create_driver
     29:     time = Time.parse("2012-01-02 13:14:15").to_i
     30:     d.run do
     31:       d.emit({'msg' => 'xyz 123', 'alt' => 'aaa bbb ccc'}, time)
=================================================================================================================================================================================================
E
=================================================================================================================================================================================================
Error: test_emit1(ForestTestOutputTest): NoMethodError: undefined method `router=' for #<Fluent::ForestTestOutput:007f9f12fb02f8>
/Users/hhatake/Github/fluent-plugin-forest/vendor/bundle/ruby/2.3.0/bundler/gems/fluentd-d5476d747846/lib/fluent/test/base.rb:53:in `initialize'
/Users/hhatake/Github/fluent-plugin-forest/vendor/bundle/ruby/2.3.0/bundler/gems/fluentd-d5476d747846/lib/fluent/test/input_test.rb:25:in `initialize'
/Users/hhatake/Github/fluent-plugin-forest/vendor/bundle/ruby/2.3.0/bundler/gems/fluentd-d5476d747846/lib/fluent/test/output_test.rb:38:in `initialize'
/Users/hhatake/Github/fluent-plugin-forest/test/plugin/test_out_forest_test.rb:17:in `new'
/Users/hhatake/Github/fluent-plugin-forest/test/plugin/test_out_forest_test.rb:17:in `create_driver'
/Users/hhatake/Github/fluent-plugin-forest/test/plugin/test_out_forest_test.rb:51:in `test_emit1'
     48:   end
     49: 
     50:   def test_emit1
  => 51:     d = create_driver
     52:     time = Time.parse("2012-01-02 13:14:15").to_i
     53:     d.run do
     54:       d.emit({'msg' => 'xyz 123', 'alt' => 'aaa bbb ccc'}, time)
=================================================================================================================================================================================================
E
=================================================================================================================================================================================================
Error: test_emit2(ForestTestOutputTest): NoMethodError: undefined method `router=' for #<Fluent::ForestTestOutput:007f9f12fd3668>
/Users/hhatake/Github/fluent-plugin-forest/vendor/bundle/ruby/2.3.0/bundler/gems/fluentd-d5476d747846/lib/fluent/test/base.rb:53:in `initialize'
/Users/hhatake/Github/fluent-plugin-forest/vendor/bundle/ruby/2.3.0/bundler/gems/fluentd-d5476d747846/lib/fluent/test/input_test.rb:25:in `initialize'
/Users/hhatake/Github/fluent-plugin-forest/vendor/bundle/ruby/2.3.0/bundler/gems/fluentd-d5476d747846/lib/fluent/test/output_test.rb:38:in `initialize'
/Users/hhatake/Github/fluent-plugin-forest/test/plugin/test_out_forest_test.rb:17:in `new'
/Users/hhatake/Github/fluent-plugin-forest/test/plugin/test_out_forest_test.rb:17:in `create_driver'
/Users/hhatake/Github/fluent-plugin-forest/test/plugin/test_out_forest_test.rb:77:in `test_emit2'
     74:   end
     75: 
     76:   def test_emit2
  => 77:     d = create_driver
     78:     time = Time.parse("2012-01-02 13:14:15").to_i
     79:     d.run do
     80:       d.emit({'msg' => 'xyz 123', 'alt' => 'aaa bbb ccc'}, time)
=================================================================================================================================================================================================
```

**Comment by tagomoris** [comment]
It occurs at here: https://github.com/tagomoris/fluent-plugin-secure-forward/blob/master/test/helper.rb#L16
`Fluent::Log` should be initialized with parent Logger object and its level, not device for logger.
So it's bug of that code, not Fluentd itself.

(It's a kind of bug, brought by non-well-prepared logging for test environment. And it's resolved in v0.14.)
